File: outbox/poll-based/orderservice/src/main/java/com/varunu28/orderservice/controller/OrderController.java
package com.varunu28.orderservice.controller;

import com.varunu28.orderservice.dto.CreateOrderDto;
import com.varunu28.orderservice.dto.OrderDto;
import com.varunu28.orderservice.service.OrderService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/orders")
public class OrderController {

    private final OrderService orderService;

    public OrderController(OrderService orderService) {
        this.orderService = orderService;
    }

    @PostMapping
    public ResponseEntity<OrderDto> createOrder(@Valid @RequestBody CreateOrderDto request) {
        return ResponseEntity.status(HttpStatus.CREATED)
            .body(orderService.createOrder(request));
    }
}

File: saga/order-service/src/main/kotlin/com/saga/orderservice/OrderController.kt
package com.saga.orderservice

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import java.math.BigInteger

@RestController
@RequestMapping("/api/v1/orders")
class OrderController(private val service: OrderService) {

    @PostMapping
    fun createOrder(@RequestBody request: CreateOrderRequest): ResponseEntity<BigInteger> {
        val orderId = service.createOrder(request)
        return ResponseEntity.status(HttpStatus.CREATED).body(orderId)
    }

    @GetMapping("/{orderId}")
    fun getOrder(@PathVariable orderId: BigInteger): ResponseEntity<Order> {
        return ResponseEntity.status(HttpStatus.OK).body(service.getOrderById(orderId))
    }
}